tests/src/Context/MessageContext.php

```diff
@@ -28,7 +29,276 @@ public function iShouldSeeThisFollowingErrorMessage(PyStringNode $message) {
     // Implode PyStringNode item because assertErrorVisible make its test
     // on the text cleaned of its HTML tags.
     $message_to_test = implode(' ', $message->getStrings());
-    parent::assertErrorVisible($message_to_test);
+    $this->assertErrorVisible($message_to_test);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertMessage($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertMessage($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_message_selector');
+      $this->assert(
+        $message,
+        $selector,
+        "The page '%s' does not contain any messages",
+        "The page '%s' does not contain the message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertNotMessage($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertNotMessage($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_message_selector');
+      $this->assertNot(
+        $message,
+        $selector,
+        "The page '%s' contains the message '%s'"
+      );
+    }
+  }
+
+  /**
+   * Checks if the current page contains the given error message.
+   *
+   * @param string $message
+   *   string The text to be checked.
+   *
+   * @Then I should see the error message( containing) :message in the Media modal
+   */
+  public function assertMediaErrorVisible($message) {
+    parent::assertErrorVisible($message);
+  }
+
+  /**
+   * Checks if the current page does not contain the given error message.
+   *
+   * @param string $message
+   *   string The text to be checked.
+   *
+   * @Given I should not see the error message( containing) :message in the Media modal
+   */
+  public function assertMediaNotErrorVisible($message) {
+    parent::assertNotErrorVisible($message);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertErrorVisible($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertErrorVisible($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_error_message_selector');
+      $this->assert(
+        $message,
+        $selector,
+        "The page '%s' does not contain any error messages",
+        "The page '%s' does not contain the error message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
```

Comment:
I will change but do not use a Drupal reference in project that does not concern Drupal directly.